File: nxflix-agents-python/src/nxflix_agents/models/content_type.py
"""Content type enum for JLPT N1 study materials."""

from enum import Enum


class ContentType(str, Enum):
    """Types of JLPT N1 study content."""

    GRAMMAR = "grammar"
    VOCABULARY = "vocabulary"
    KANJI = "kanji"
    READING = "reading"
    LISTENING = "listening"

    @classmethod
    def is_valid(cls, value: str) -> bool:
        """Check if a string is a valid content type."""
        try:
            cls(value)
            return True
        except ValueError:
            return False

File: nxflix-agents-ts/src/models/content-type.ts
import { z } from 'zod';

/**
 * Content types for JLPT N1 study materials.
 */
export const ContentType = z.enum([
  'grammar',
  'vocabulary',
  'kanji',
  'reading',
  'listening',
]);
export type ContentType = z.infer<typeof ContentType>;

/**
 * Helper to check if a string is a valid content type.
 */
export function isValidContentType(value: string): value is ContentType {
  return ContentType.safeParse(value).success;
}
